pkg/resource/predicates.go

```diff
@@ -164,13 +166,64 @@ func IsNamed(name string) PredicateFn {
 // DesiredStateChanged accepts objects that have changed their desired state, i.e.
 // the state that is not managed by the controller.
 // To be more specific, it accepts update events that have changes in one of the followings:
-// - `metadata.annotations`
+// - `metadata.annotations` (except for certain annotations)
 // - `metadata.labels`
 // - `spec`
 func DesiredStateChanged() predicate.Predicate {
 	return predicate.Or(
-		predicate.AnnotationChangedPredicate{},
+		AnnotationChangedPredicate{
+			ignored: []string{
+				// These annotations are managed by the controller and should
+				// not be considered as a change in desired state. The managed
+				// reconciler explicitly requests a new reconcile already after
+				// updating these annotations.
+				meta.AnnotationKeyExternalCreateFailed,
+				meta.AnnotationKeyExternalCreatePending,
```

Comment:
I'm not sure about hard coding these ignored annotations. Would it make sense to pass them in as arguments to `DesiredStateChanged`?

My thinking is:

* What if we want to use `DesiredStateChanged` in another context in future?
* What if we want to ignore other annotations in future, but not for 100% of callers?

Does every MR reconciler `Setup` call need to call `DesiredStateChanged` to create new predicates? If so I could understand if we didn't want to have to update all that boilerplate in future but I'm still on the fence.

Reply:
> Does every MR reconciler Setup call need to call DesiredStateChanged to create new predicates? 

Yes, please see as an example: https://github.com/crossplane-contrib/provider-aws/blob/44b8d0f3bb46c179aa5c69e0a2c13dfeb6f4d863/pkg/controller/database/dbsubnetgroup/controller.go#L71

This is the main reason why I didn't want to introduce them as arguments. 

> - What if we want to use DesiredStateChanged in another context in future?
> - What if we want to ignore other annotations in future, but not for 100% of callers?

Once we have some concrete use cases like these, we can introduce some optional arguments where the default behavior can be overridden.